### PR TITLE
Display an error when a user is not allowed to view a course

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
When following a link to a course you are not enrolled in, you currently just get an empty page.

### Description
Now displays an error message:
grafik

### Steps for Testing

1. Log in to Artemis using an account that does have access rights to a course
2. Navigate to the course view page (e.g. "https://vmbruegge60.in.tum.de/#/courses/1867/" on TS3)
3. Check you can still access the page normally
4. Log out and then in again using a different account without access rights (e.g. artemis_test_user_10 on TS3)
5. Try to access the same URL ("https://vmbruegge60.in.tum.de/#/courses/1867/") directly (paste into browser bar)
6. Check if an error message appears

### CODE REVIEW

**Comment by FrankeLukas** [comment]
I suggest reviewing using https://github.com/ls1intum/Artemis/pull/2355/files?w=1 to hide the white-space-only changes.

**Review by tiffca — APPROVED** [review]
Works as described (tested on test server 3), confirmed user sees the error message when they're not assigned to that course. Courses show up fine otherwise.

**Review by schultek — APPROVED** [review]
This works and the code looks good.

**Review by derLalla — CHANGES_REQUESTED** [review]
image

I am logged in as testuser3 and while accessing a not allowed page (https://vmbruegge60.in.tum.de/#/courses/1610/exercises) I am not getting the error message. I am only getting errors in the console and network tab.

File: src/main/webapp/app/overview/course-overview.component.ts [hunk 1]
```diff
@@ -61,12 +63,19 @@ export class CourseOverviewComponent implements OnInit, OnDestroy {
...
+            (res: HttpResponse<Course>) => {
+                this.courseCalculationService.updateCourse(res.body!);
+                this.course = this.courseCalculationService.getCourse(this.courseId);
+                this.adjustCourseDescription();
+                setTimeout(() => (this.refreshingCourse = false), 500); // ensure min animation duration
+            },
+            (error: HttpErrorResponse) => {
+                const errorMessage = error.headers.get('X-artemisApp-message')!;
+                const jhiAlert = this.jhiAlertService.error(errorMessage);
+                jhiAlert.msg = errorMessage;
```
Comment (stefanwaldhauser) [inline src/main/webapp/app/overview/course-overview.component.ts:76]: I am still not quite sure what this line does as you passed the alert service the errorMessage already.

**Review by stefanwaldhauser — APPROVED** [review]
Tested it on TS1 with this course: https://artemistest.ase.in.tum.de/#/courses/298 where artemis_test_user_2 is not a student. I got the following error message;
image
image
Nice Work :)

**Comment by FrankeLukas** [comment]
@derLalla I don't know what went wrong, but I can view the course you linked using `artemis_test_user_3` without problems. Seeing the 401 error for `/api/account` leads me to believe something unrelated broke.

File: src/main/webapp/app/overview/course-overview.component.ts (hunk 1 above)
Comment (FrankeLukas) [reply src/main/webapp/app/overview/course-overview.component.ts:76]: I actually don't know either, but seeing this used in other places I thought it covers some edge case I have not thought of yet. Should I remove it?

**Review by derLalla — APPROVED** [review]
it's working as expected on ts2. So I don't know what went wrong..

**Review by krusche — APPROVED** [review]

**Pull request merged**